Source/WebKit/GPUProcess/graphics/WebGPU/RemoteAdapter.cpp

```diff
@@ -66,7 +66,13 @@ void RemoteAdapter::requestDevice(const WebGPU::DeviceDescriptor& descriptor, We
         return;
     }
 
-    m_backing->requestDevice(*convertedDescriptor, [callback = WTFMove(callback), objectHeap = Ref { m_objectHeap }, streamConnection = m_streamConnection.copyRef(), identifier, queueIdentifier] (Ref<PAL::WebGPU::Device>&& device) mutable {
+    m_backing->requestDevice(*convertedDescriptor, [callback = WTFMove(callback), objectHeap = Ref { m_objectHeap }, streamConnection = m_streamConnection.copyRef(), identifier, queueIdentifier] (RefPtr<PAL::WebGPU::Device>&& devicePtr) mutable {
+        if (!devicePtr.get()) {
```

Comment:
Ditto about invalid device vs nullptr. Generally, things shouldn't be null, but should be invalid instead. Are we super confident that null is actually correct here?

Reply:
Fairly certain null is correct. The device is never returned since the promise is rejected. If we created an invalid deice, what would we do with it? We don't return it via the promise